src/main/java/com/bookstore/exception/AuthenticationException.java

```diff
@@ -0,0 +1,11 @@
+package com.bookstore.exception;
+
+public class AuthenticationException extends RuntimeException {
```

Comment:
How are you goind to handle this exception? It still throws in filter and won't be handled by GlobalHandler. 
If you check it and it works - leave as it is